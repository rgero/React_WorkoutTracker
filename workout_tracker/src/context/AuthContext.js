import AsyncStorage from '@react-native-async-storage/async-storage';

import trackerAPI from '../api/workouttracker';
import createDataContext from "./createDataContext";

const authReducer = (state, action) => {
    switch (action.type)
    {
        case "addError":
            return { ...state, errorMessage: action.payload};
        case "storeToken":
            return { errorMessage: "", token: action.payload};
        case "clearErrorMessage":
            return {...state, errorMessage: ""}
        case "signout":
            return { token: null, errorMessage: ""}
        default:
            return state;
    }
};

const tryLocalSignin = dispatch => async () => {
    const token = await AsyncStorage.getItem('token');
    if (token)
    {
        dispatch({ type: 'storeToken', payload: token})
    }
}

const clearErrorMessage = dispatch => () => {
    dispatch( {type: 'clearErrorMessage' })
}

const signUp = dispatch => {
    return async ({ email, displayName, password }) => {
        try {
            const response = await trackerAPI.post('/signup', { email, displayName, password });
            
            // Storing the Token
            await AsyncStorage.setItem('token', response.data.token);
            dispatch({type: "storeToken", payload: response.data.token})
        } catch (err) {
            console.log(err.message);
            dispatch({ type: 'addError', payload: "Sign-up failed. Please try again"})
        }
    };
}

const signIn = (dispatch) => {
    return async ({ email, password }) => {
        try {
            const response = await trackerAPI.post('/signin', { email, password });
            
            // Storing the Token
            await AsyncStorage.setItem('token', response.data.token);
            dispatch({type: "storeToken", payload: response.data.token})
        } catch (err) {
            console.log(err.message);
            dispatch({ type: 'addError', payload: "Login failed. Please try again"})
        }
    };
}

const signOut = dispatch => async () => {
    try {
        await AsyncStorage.removeItem('token');
        dispatch({type: "signout"})
    } catch (err)
    {
        console.log(err);
    }
}

export const {Provider, Context } = createDataContext(authReducer, 
                                                      {clearErrorMessage, signUp, signIn, signOut, tryLocalSignin}, 
                                                      {
                                                        token: null,
                                                        errorMessage: ""
                                                      }
                                                    )