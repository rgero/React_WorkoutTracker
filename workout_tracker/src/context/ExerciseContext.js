import trackerAPI from '../api/workouttracker';
import createDataContext from "./createDataContext";

const exerciseReducer = (state, action) => {
    switch(action.type)
    {
        case 'fetch':
            return action.payload;
        default:
            return state;
    }
};

const fetchExercises = dispatch => async () => {
    const response = await trackerAPI.get('/exercises');
    dispatch({type: 'fetch', payload: response.data})
};

const fetchExerciseByID = dispatch => async (targetID) => {
    const response = await trackerAPI.get(`/exercises/${targetID}`);
    dispatch({type: 'fetch', payload: response.data})
};


export const {Provider, Context} = createDataContext(
    exerciseReducer,
    {fetchExercises, fetchExerciseByID},
    []
);