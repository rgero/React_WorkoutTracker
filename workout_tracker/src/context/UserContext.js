import AsyncStorage from '@react-native-async-storage/async-storage';

import trackerAPI from '../api/workouttracker';
import createDataContext from "./createDataContext";

const userReducer = (state, action) => {
    switch(action.type)
    {
        case 'fetch':
            return action.payload;
        default:
            return state;
    }
};

const getUser = dispatch => async () => {
    const token = await AsyncStorage.getItem('token');
    if (token)
    {
        const response = await trackerAPI.get('/users');
        dispatch({type: 'fetch', payload: response.data})
    }
}

export const {Provider, Context} = createDataContext(
    userReducer,
    {getUser},
    []
);