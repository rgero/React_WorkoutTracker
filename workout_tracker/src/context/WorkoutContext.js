import trackerAPI from '../api/workouttracker';
import createDataContext from "./createDataContext";

const workoutReducer = (state, action) => {
    switch(action.type)
    {
        case 'fetch':
            return action.payload;
        case "create":
            return state.concat(action.payload);
        case "edit":
            return state.map( (workout) => {
                if (workout._id === action._id) {
                    return {
                        ...workout,
                        ...action.update
                    }
                } else {
                    return workout;
                }
            })
        case "delete":
            return state.filter( (testWorkout) => {
                return action.id !== testWorkout._id;
            });
        default:
            return state;
    }
};

const fetchWorkouts = dispatch => async () => {
    const response = await trackerAPI.get('/workouts');
    dispatch({type: 'fetch', payload: response.data})
};

const createWorkout = dispatch => async (workout) => 
{
    const response = await trackerAPI.post('/workouts', workout)
    dispatch({type: "create", payload: response.data});
};

const updateWorkout = dispatch => async (workout) =>
{
    const response = await trackerAPI.put(`/workouts/${workout._id}`, workout);
    dispatch({type: "edit", payload: response.data});
}

const deleteWorkout = dispatch => async (targetID) => 
{
    const response = await trackerAPI.delete(`/workouts/${targetID}`)
    dispatch({type: "delete", id: targetID});
}

export const {Provider, Context} = createDataContext(
    workoutReducer,
    {fetchWorkouts, createWorkout, deleteWorkout, updateWorkout},
    []
);