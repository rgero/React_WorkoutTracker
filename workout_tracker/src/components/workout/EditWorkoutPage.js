import React, { useContext, useEffect, useState } from 'react';
import {useParams} from "react-router-dom";
import Container from 'react-bootstrap/Container';

import {Context as WorkoutContext} from '../../context/WorkoutContext';
import { WorkoutForm } from './WorkoutForm';

export const EditWorkoutPage = ( )=> 
{
    const {state, fetchWorkoutByID, updateWorkout} = useContext(WorkoutContext);
    const [isLoaded, setLoaded] = useState(false);
    const { id } = useParams();

    useEffect(()=> {
        const GetWorkoutByID = async (id) => {
            await fetchWorkoutByID(id);
            setLoaded(true);
        }
        GetWorkoutByID(id);
    }, []);

    if (state.workoutDate)
    {
        return (
            <Container fluid="md">
                <h1>Edit a workout</h1>
                <WorkoutForm
                    workout={state}
                    errorMessage={state.errorMessage}
                    onSubmit={(workout)=> updateWorkout(workout)}
                />
            </Container>
            )
    } else {
        return (<div>Loading</div>)
    }
}