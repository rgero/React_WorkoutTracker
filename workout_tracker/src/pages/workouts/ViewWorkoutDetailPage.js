import React from 'react';
import {useNavigate, useParams} from "react-router-dom";

import Alert from 'react-bootstrap/Alert';
import Container from 'react-bootstrap/Container';
import Col from 'react-bootstrap/Col';
import Row from 'react-bootstrap/Row';
import Button from 'react-bootstrap/Button';
import Card from 'react-bootstrap/Card';
import Dropdown from 'react-bootstrap/Dropdown';

import DateFormatter from '../../helpers/DateFormatter';
import {Context as WorkoutContext} from '../../context/WorkoutContext';
import { ExerciseList } from '../../components/exercise/ExerciseList';

import { confirmAlert } from 'react-confirm-alert'; // Import
import 'react-confirm-alert/src/react-confirm-alert.css'; // Import css

const ViewWorkoutDetailsPage = ()=> 
{
    const navigate = useNavigate();
    const { id } = useParams();

    const {state, fetchWorkouts, deleteWorkout} = React.useContext(WorkoutContext);
    const [currentWorkout, setCurrentWorkout] = React.useState({});
    const [isLoaded, setLoaded] = React.useState(false);
    const [error, setError] = React.useState("");


    const processDelete = async () => {
        confirmAlert({
            title: 'Confirm to delete?',
            message: 'Are you sure to do this.',
            buttons: [
              {
                label: 'Yes',
                onClick: async () => {
                    await deleteWorkout(currentWorkout._id);
                    navigate('/dashboard')
                }
              },
              {
                label: 'No'
              }
            ]
          });

    }

    const isValidWorkout = () => {
        return !Object.keys(currentWorkout).length == 0;
    }

    React.useEffect(()=> {
        const GetWorkoutByID = async (id) => {
            if (!isLoaded)
            {
                await fetchWorkouts();
                setLoaded(true);
            }
            let loadedWorkout = state.filter( (targetWorkout) => {
                return id === targetWorkout._id;
            });

            if (loadedWorkout.length == 0)
            {
                setError("An error has occurred");
            } else {
                setError("");
                setCurrentWorkout(loadedWorkout[0]);
            }
        }
        GetWorkoutByID(id);
    }, [fetchWorkouts, id, isLoaded, state]);

    if (isLoaded) {
        if (isValidWorkout()) { 
            return (
                <Container className="pt-4" fluid="md">
                    <Row className="align-items-center">
                        <Col>
                            <h1>Workout on {DateFormatter(currentWorkout.workoutDate)}</h1> 
                        </Col>
                        <Col>
                            <Dropdown className="float-end">
                                <Dropdown.Toggle id="dropdown-basic">
                                    Options
                                </Dropdown.Toggle>
                                <Dropdown.Menu>
                                    <Dropdown.Item href={`/edit/${currentWorkout._id}`}>Edit</Dropdown.Item>
                                    <Dropdown.Item onClick={processDelete}>Delete</Dropdown.Item>
                                </Dropdown.Menu>
                            </Dropdown>
                        </Col>
                    </Row>
                    <Row>
                        {currentWorkout.notes !== "" ? (
                            <Card>
                                <Card.Body>
                                    <Card.Title>Notes</Card.Title>
                                    <Card.Body>
                                        {currentWorkout.notes}
                                    </Card.Body>
                                </Card.Body>
                            </Card>
                        ) : ( null ) }
                        <Card>
                            <Card.Body>
                                <Card.Title>Exercises - {currentWorkout.exerciseList.length}</Card.Title>
                                <Card.Body>
                                    <ExerciseList exerciseList={currentWorkout.exerciseList}/>
                                </Card.Body>
                                <Card.Body>
                                    
                                </Card.Body>
                            </Card.Body>
                        </Card>
                        <Col className="pt-2"><Button className="float-end" href="/dashboard">Return to Dashboard</Button></Col>
                    </Row>
                </Container>
            )
        } else if (error)
        {
            return(
                <Container>
                    <Alert key="danger" variant="danger">
                        <Alert.Heading>Oh snap! You got an error!</Alert.Heading>
                        <p>{error}</p>
                        <div className="d-flex justify-content-end">
                            <Button onClick={() => navigate('/dashboard')} variant="outline-danger">
                                Return to Dashboard
                            </Button>
                        </div>
                    </Alert>
                    
                </Container>
            )
        }
    } else {
        return (
            <Container>
                Loading
            </Container>
        )
    }
}

export default ViewWorkoutDetailsPage;