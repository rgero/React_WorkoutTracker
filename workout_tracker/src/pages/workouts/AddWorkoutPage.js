import React from 'react';
import Container from 'react-bootstrap/Container';

import {Context as WorkoutContext} from '../../context/WorkoutContext';
import { WorkoutForm } from '../../components/workout/WorkoutForm';

export const AddWorkoutPage = ()=> 
{
    const {state, createWorkout} = React.useContext(WorkoutContext);
    return (
        <Container fluid="md">
            <h1>Create a new workout</h1>
            <WorkoutForm
                onSubmit={(workout)=> createWorkout(workout)}
            />
        </Container>
    )
}