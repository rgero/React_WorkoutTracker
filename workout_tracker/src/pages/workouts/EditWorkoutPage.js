import React from 'react';
import {useParams} from "react-router-dom";
import Container from 'react-bootstrap/Container';

import {Context as WorkoutContext} from '../../context/WorkoutContext';
import { WorkoutForm } from '../../components/workout/WorkoutForm';

export const EditWorkoutPage = ( )=> 
{
    const {state, fetchWorkouts, updateWorkout} = React.useContext(WorkoutContext);
    const [isLoaded, setLoaded] = React.useState(false);
    const [currentWorkout, setCurrentWorkout] = React.useState({});
    const [error, setError] = React.useState("");
    const { id } = useParams();

    React.useEffect(()=> {
        const GetWorkoutByID = async (id) => {
            if (!isLoaded)
            {
                await fetchWorkouts();
                setLoaded(true);
            }

            let loadedWorkout = state.filter( (targetWorkout) => {
                return id === targetWorkout._id;
            });
                
            if (loadedWorkout.length == 0)
            {
                setError("An error has occurred");
            } else {
                setError("");
                setCurrentWorkout(loadedWorkout[0]);
            }
        }
        GetWorkoutByID(id);
    }, [state]);

    const isValidWorkout = () => {
        return !Object.keys(currentWorkout).length == 0;
    }

    if (isLoaded) {
        if (isValidWorkout()) {
            return (
                <Container fluid="md">
                    <h1>Edit a workout</h1>
                    <WorkoutForm
                        workout={currentWorkout}
                        onSubmit={(workout)=> updateWorkout(workout)}
                    />
                </Container>
                )
        } else if (error)
        {
            return(
                <Container>
                    {error}
                </Container>
            )
        }
    } else {
        return (<div>Loading</div>)
    }
}