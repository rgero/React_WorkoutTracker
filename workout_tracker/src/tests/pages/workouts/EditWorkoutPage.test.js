import React from 'react';
import TestRenderer from 'react-test-renderer';
import reactRouterDom from 'react-router-dom';
import { act } from 'react-dom/test-utils';

import {Context as WorkoutContext} from '../../../context/WorkoutContext';
import { EditWorkoutPage } from '../../../pages/workouts/EditWorkoutPage';
import {TestWorkouts} from '../../testData';

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useParams: jest.fn(),
    useNavigate: jest.fn()
}));

test('Rendering a Workout', async ()=> {

    const testContextValues = {
        state: TestWorkouts,
        fetchWorkouts: jest.fn(),
        updateWorkouts: jest.fn()
    }

    jest.spyOn(reactRouterDom, 'useParams').mockReturnValue({ id: '1234' });

    let element;
    await act(async () => {
        element = new TestRenderer.create(
            <WorkoutContext.Provider value={testContextValues}>
                <EditWorkoutPage/>
            </WorkoutContext.Provider>
        );
    })
    
    expect(element.toJSON()).toMatchSnapshot();
})

test('Rendering a Second Workout', async ()=> {

    const testContextValues = {
        state: TestWorkouts,
        fetchWorkouts: jest.fn(),
        updateWorkouts: jest.fn()
    }

    jest.spyOn(reactRouterDom, 'useParams').mockReturnValue({ id: '2222' });

    let element;
    await act(async () => {
        element = new TestRenderer.create(
            <WorkoutContext.Provider value={testContextValues}>
                <EditWorkoutPage/>
            </WorkoutContext.Provider>
        );
    })
    
    expect(element.toJSON()).toMatchSnapshot();
})