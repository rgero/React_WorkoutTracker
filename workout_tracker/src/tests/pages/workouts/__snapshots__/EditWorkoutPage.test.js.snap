// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering a Second Workout 1`] = `
<div
  className="container"
>
  <h1>
    Edit a workout
  </h1>
  <div
    className="container-md"
  >
    <div
      className="justify-content-md-center row"
    >
      <div
        className="col-md-8"
      >
        <div
          className="card"
        >
          <div
            className="card-header"
          >
            <h4>
              Exercise Information
            </h4>
          </div>
          <div
            className="card-body"
          >
            <form
              className=""
              id="addWorkout"
              onSubmit={[Function]}
            >
              <div
                className="mb-3 form-floating"
              >
                <input
                  aria-label="workoutDate"
                  className="form-control"
                  id="floatingWorkoutDate"
                  onChange={[Function]}
                  type="date"
                  value={"2022-02-08T00:00:00.000Z"}
                />
                <label
                  htmlFor="floatingWorkoutDate"
                >
                  Workout Date
                </label>
              </div>
              <div
                className="mb-3 form-floating"
              >
                <textarea
                  aria-label="workoutNotes"
                  className="form-control"
                  id="floatingNotes"
                  onChange={[Function]}
                  rows={3}
                  value="February"
                />
                <label
                  htmlFor="floatingNotes"
                >
                  Notes
                </label>
              </div>
            </form>
          </div>
        </div>
        <div
          className="card"
        >
          <div
            className="card-header"
          >
            <h4>
              Add/Edit Exercises
            </h4>
          </div>
          <div
            className="card-body"
          >
            <div
              className="container-md"
            >
              <div
                className="row"
              >
                <form
                  className=""
                  id="addExerciseForm"
                  onSubmit={[Function]}
                >
                  <div
                    className="mb-3 form-floating"
                  >
                    <input
                      aria-label="name"
                      className="form-control"
                      id="floatingExerciseName"
                      onChange={[Function]}
                      type="text"
                      value=""
                    />
                    <label
                      htmlFor="floatingExerciseName"
                    >
                      Name
                    </label>
                  </div>
                  <div
                    className="mb-3 form-floating"
                  >
                    <input
                      aria-label="muscleGroup"
                      className="form-control"
                      id="floatingMuscleGroup"
                      onChange={[Function]}
                      type="text"
                      value=""
                    />
                    <label
                      htmlFor="floatingMuscleGroup"
                    >
                      Muscle Group
                    </label>
                  </div>
                  <div
                    className="mb-3 form-floating"
                  >
                    <textarea
                      aria-label="exerciseNotes"
                      className="form-control"
                      id="floatingExerciseNotes"
                      onChange={[Function]}
                      rows={3}
                      value=""
                    />
                    <label
                      htmlFor="floatingExerciseNotes"
                    >
                      Notes
                    </label>
                  </div>
                </form>
              </div>
              <div
                className="row"
              >
                <div
                  className="container-md"
                >
                  <form
                    className=""
                    onSubmit={[Function]}
                  >
                    <div
                      className="justify-content-md-center align-items-md-center row row-cols-md-4"
                    >
                      <div
                        className="col-md"
                      >
                        <div
                          className="mb-3 form-floating"
                        >
                          <input
                            aria-label="reps"
                            className="form-control"
                            id="floatingReps"
                            onChange={[Function]}
                            type="text"
                            value=""
                          />
                          <label
                            htmlFor="floatingReps"
                          >
                            Reps
                          </label>
                        </div>
                      </div>
                      <div
                        className="col-md"
                      >
                        <div
                          className="mb-3 form-floating"
                        >
                          <input
                            aria-label="weight"
                            className="form-control"
                            id="floatingWeight"
                            onChange={[Function]}
                            type="text"
                            value=""
                          />
                          <label
                            htmlFor="floatingWeight"
                          >
                            Weight
                          </label>
                        </div>
                      </div>
                    </div>
                    <button
                      aria-label="setSubmit"
                      className="float-end btn btn-outline-secondary"
                      disabled={false}
                      type="submit"
                    >
                      Add Set
                    </button>
                  </form>
                </div>
              </div>
              <div
                className="row"
              >
                <button
                  aria-label="exerciseSubmit"
                  className="mt-4 btn btn-outline-secondary"
                  disabled={false}
                  form="addExerciseForm"
                  type="submit"
                >
                  Add Exercise
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          className="accordion"
        >
          <div
            className="accordion-item"
          >
            <h2
              className="accordion-header"
            >
              <button
                aria-expanded={false}
                className="accordion-button collapsed"
                onClick={[Function]}
                type="button"
              >
                Exercise List
              </button>
            </h2>
            <div
              aria-expanded={null}
              className="accordion-collapse collapse"
            >
              <div
                className="accordion-body"
              >
                <div
                  className="container"
                >
                  <div
                    className="list-group list-group-flush"
                    onKeyDown={[Function]}
                    style={
                      Object {
                        "margin": "auto",
                        "width": "60%",
                      }
                    }
                  >
                    <button
                      className="list-group-item list-group-item-action"
                      data-rr-ui-event-key={null}
                      onClick={[Function]}
                      style={
                        Object {
                          "border": "0px",
                        }
                      }
                    >
                      <div
                        className="card"
                        onClick={[Function]}
                      >
                        <div
                          className="card-body"
                        >
                          <div
                            className="card-title h5"
                          >
                            #
                            1
                             - 
                            Incline Press
                          </div>
                          <div
                            className="card-body"
                          >
                            <div
                              className="pt-3 container"
                              style={
                                Object {
                                  "margin": "auto",
                                  "textAlign": "center",
                                  "width": "60%",
                                }
                              }
                            >
                              <div
                                style={
                                  Object {
                                    "textAlign": "left",
                                  }
                                }
                              >
                                Set List
                              </div>
                              <table
                                className="table table-sm table-striped table-bordered table-hover"
                              >
                                <thead>
                                  <tr>
                                    <th>
                                      Reps
                                    </th>
                                    <th>
                                      Weight
                                    </th>
                                  </tr>
                                </thead>
                                <tbody>
                                  <tr
                                    onClick={[Function]}
                                  >
                                    <td>
                                      10
                                    </td>
                                    <td>
                                      275
                                    </td>
                                  </tr>
                                  <tr
                                    onClick={[Function]}
                                  >
                                    <td>
                                      10
                                    </td>
                                    <td>
                                      300
                                    </td>
                                  </tr>
                                  <tr
                                    onClick={[Function]}
                                  >
                                    <td>
                                      3
                                    </td>
                                    <td>
                                      315
                                    </td>
                                  </tr>
                                </tbody>
                              </table>
                            </div>
                          </div>
                        </div>
                      </div>
                    </button>
                    <button
                      className="list-group-item list-group-item-action"
                      data-rr-ui-event-key={null}
                      onClick={[Function]}
                      style={
                        Object {
                          "border": "0px",
                        }
                      }
                    >
                      <div
                        className="card"
                        onClick={[Function]}
                      >
                        <div
                          className="card-body"
                        >
                          <div
                            className="card-title h5"
                          >
                            #
                            2
                             - 
                            Lunges
                          </div>
                          <div
                            className="card-body"
                          >
                            <div
                              className="pt-3 container"
                              style={
                                Object {
                                  "margin": "auto",
                                  "textAlign": "center",
                                  "width": "60%",
                                }
                              }
                            >
                              <div
                                style={
                                  Object {
                                    "textAlign": "left",
                                  }
                                }
                              >
                                Set List
                              </div>
                              <table
                                className="table table-sm table-striped table-bordered table-hover"
                              >
                                <thead>
                                  <tr>
                                    <th>
                                      Reps
                                    </th>
                                    <th>
                                      Weight
                                    </th>
                                  </tr>
                                </thead>
                                <tbody>
                                  <tr
                                    onClick={[Function]}
                                  >
                                    <td>
                                      40
                                    </td>
                                    <td>
                                      100
                                    </td>
                                  </tr>
                                  <tr
                                    onClick={[Function]}
                                  >
                                    <td>
                                      40
                                    </td>
                                    <td>
                                      100
                                    </td>
                                  </tr>
                                  <tr
                                    onClick={[Function]}
                                  >
                                    <td>
                                      40
                                    </td>
                                    <td>
                                      100
                                    </td>
                                  </tr>
                                  <tr
                                    onClick={[Function]}
                                  >
                                    <td>
                                      40
                                    </td>
                                    <td>
                                      100
                                    </td>
                                  </tr>
                                </tbody>
                              </table>
                            </div>
                          </div>
                        </div>
                      </div>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <button
          aria-label="workoutSubmit"
          className="btn btn-outline-secondary"
          disabled={false}
          form="addWorkout"
          type="submit"
        >
          Submit Workout
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Rendering a Workout 1`] = `
<div
  className="container"
>
  <h1>
    Edit a workout
  </h1>
  <div
    className="container-md"
  >
    <div
      className="justify-content-md-center row"
    >
      <div
        className="col-md-8"
      >
        <div
          className="card"
        >
          <div
            className="card-header"
          >
            <h4>
              Exercise Information
            </h4>
          </div>
          <div
            className="card-body"
          >
            <form
              className=""
              id="addWorkout"
              onSubmit={[Function]}
            >
              <div
                className="mb-3 form-floating"
              >
                <input
                  aria-label="workoutDate"
                  className="form-control"
                  id="floatingWorkoutDate"
                  onChange={[Function]}
                  type="date"
                  value={"2023-03-08T00:00:00.000Z"}
                />
                <label
                  htmlFor="floatingWorkoutDate"
                >
                  Workout Date
                </label>
              </div>
              <div
                className="mb-3 form-floating"
              >
                <textarea
                  aria-label="workoutNotes"
                  className="form-control"
                  id="floatingNotes"
                  onChange={[Function]}
                  rows={3}
                  value="Testing Notes for Testing"
                />
                <label
                  htmlFor="floatingNotes"
                >
                  Notes
                </label>
              </div>
            </form>
          </div>
        </div>
        <div
          className="card"
        >
          <div
            className="card-header"
          >
            <h4>
              Add/Edit Exercises
            </h4>
          </div>
          <div
            className="card-body"
          >
            <div
              className="container-md"
            >
              <div
                className="row"
              >
                <form
                  className=""
                  id="addExerciseForm"
                  onSubmit={[Function]}
                >
                  <div
                    className="mb-3 form-floating"
                  >
                    <input
                      aria-label="name"
                      className="form-control"
                      id="floatingExerciseName"
                      onChange={[Function]}
                      type="text"
                      value=""
                    />
                    <label
                      htmlFor="floatingExerciseName"
                    >
                      Name
                    </label>
                  </div>
                  <div
                    className="mb-3 form-floating"
                  >
                    <input
                      aria-label="muscleGroup"
                      className="form-control"
                      id="floatingMuscleGroup"
                      onChange={[Function]}
                      type="text"
                      value=""
                    />
                    <label
                      htmlFor="floatingMuscleGroup"
                    >
                      Muscle Group
                    </label>
                  </div>
                  <div
                    className="mb-3 form-floating"
                  >
                    <textarea
                      aria-label="exerciseNotes"
                      className="form-control"
                      id="floatingExerciseNotes"
                      onChange={[Function]}
                      rows={3}
                      value=""
                    />
                    <label
                      htmlFor="floatingExerciseNotes"
                    >
                      Notes
                    </label>
                  </div>
                </form>
              </div>
              <div
                className="row"
              >
                <div
                  className="container-md"
                >
                  <form
                    className=""
                    onSubmit={[Function]}
                  >
                    <div
                      className="justify-content-md-center align-items-md-center row row-cols-md-4"
                    >
                      <div
                        className="col-md"
                      >
                        <div
                          className="mb-3 form-floating"
                        >
                          <input
                            aria-label="reps"
                            className="form-control"
                            id="floatingReps"
                            onChange={[Function]}
                            type="text"
                            value=""
                          />
                          <label
                            htmlFor="floatingReps"
                          >
                            Reps
                          </label>
                        </div>
                      </div>
                      <div
                        className="col-md"
                      >
                        <div
                          className="mb-3 form-floating"
                        >
                          <input
                            aria-label="weight"
                            className="form-control"
                            id="floatingWeight"
                            onChange={[Function]}
                            type="text"
                            value=""
                          />
                          <label
                            htmlFor="floatingWeight"
                          >
                            Weight
                          </label>
                        </div>
                      </div>
                    </div>
                    <button
                      aria-label="setSubmit"
                      className="float-end btn btn-outline-secondary"
                      disabled={false}
                      type="submit"
                    >
                      Add Set
                    </button>
                  </form>
                </div>
              </div>
              <div
                className="row"
              >
                <button
                  aria-label="exerciseSubmit"
                  className="mt-4 btn btn-outline-secondary"
                  disabled={false}
                  form="addExerciseForm"
                  type="submit"
                >
                  Add Exercise
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          className="accordion"
        >
          <div
            className="accordion-item"
          >
            <h2
              className="accordion-header"
            >
              <button
                aria-expanded={false}
                className="accordion-button collapsed"
                onClick={[Function]}
                type="button"
              >
                Exercise List
              </button>
            </h2>
            <div
              aria-expanded={null}
              className="accordion-collapse collapse"
            >
              <div
                className="accordion-body"
              >
                <div
                  className="container"
                >
                  <div
                    className="list-group list-group-flush"
                    onKeyDown={[Function]}
                    style={
                      Object {
                        "margin": "auto",
                        "width": "60%",
                      }
                    }
                  >
                    <button
                      className="list-group-item list-group-item-action"
                      data-rr-ui-event-key={null}
                      onClick={[Function]}
                      style={
                        Object {
                          "border": "0px",
                        }
                      }
                    >
                      <div
                        className="card"
                        onClick={[Function]}
                      >
                        <div
                          className="card-body"
                        >
                          <div
                            className="card-title h5"
                          >
                            #
                            1
                             - 
                            Bench Press
                          </div>
                          <div
                            className="card-body"
                          >
                            <div
                              className="pt-3 container"
                              style={
                                Object {
                                  "margin": "auto",
                                  "textAlign": "center",
                                  "width": "60%",
                                }
                              }
                            >
                              <div
                                style={
                                  Object {
                                    "textAlign": "left",
                                  }
                                }
                              >
                                Set List
                              </div>
                              <table
                                className="table table-sm table-striped table-bordered table-hover"
                              >
                                <thead>
                                  <tr>
                                    <th>
                                      Reps
                                    </th>
                                    <th>
                                      Weight
                                    </th>
                                  </tr>
                                </thead>
                                <tbody>
                                  <tr
                                    onClick={[Function]}
                                  >
                                    <td>
                                      20
                                    </td>
                                    <td>
                                      275
                                    </td>
                                  </tr>
                                  <tr
                                    onClick={[Function]}
                                  >
                                    <td>
                                      20
                                    </td>
                                    <td>
                                      300
                                    </td>
                                  </tr>
                                  <tr
                                    onClick={[Function]}
                                  >
                                    <td>
                                      3
                                    </td>
                                    <td>
                                      315
                                    </td>
                                  </tr>
                                </tbody>
                              </table>
                            </div>
                          </div>
                        </div>
                      </div>
                    </button>
                    <button
                      className="list-group-item list-group-item-action"
                      data-rr-ui-event-key={null}
                      onClick={[Function]}
                      style={
                        Object {
                          "border": "0px",
                        }
                      }
                    >
                      <div
                        className="card"
                        onClick={[Function]}
                      >
                        <div
                          className="card-body"
                        >
                          <div
                            className="card-title h5"
                          >
                            #
                            2
                             - 
                            Shrugs
                          </div>
                          <div
                            className="card-body"
                          >
                            <div
                              className="pt-3 container"
                              style={
                                Object {
                                  "margin": "auto",
                                  "textAlign": "center",
                                  "width": "60%",
                                }
                              }
                            >
                              <div
                                style={
                                  Object {
                                    "textAlign": "left",
                                  }
                                }
                              >
                                Set List
                              </div>
                              <table
                                className="table table-sm table-striped table-bordered table-hover"
                              >
                                <thead>
                                  <tr>
                                    <th>
                                      Reps
                                    </th>
                                    <th>
                                      Weight
                                    </th>
                                  </tr>
                                </thead>
                                <tbody>
                                  <tr
                                    onClick={[Function]}
                                  >
                                    <td>
                                      40
                                    </td>
                                    <td>
                                      100
                                    </td>
                                  </tr>
                                  <tr
                                    onClick={[Function]}
                                  >
                                    <td>
                                      40
                                    </td>
                                    <td>
                                      100
                                    </td>
                                  </tr>
                                  <tr
                                    onClick={[Function]}
                                  >
                                    <td>
                                      40
                                    </td>
                                    <td>
                                      100
                                    </td>
                                  </tr>
                                  <tr
                                    onClick={[Function]}
                                  >
                                    <td>
                                      40
                                    </td>
                                    <td>
                                      100
                                    </td>
                                  </tr>
                                </tbody>
                              </table>
                            </div>
                          </div>
                        </div>
                      </div>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <button
          aria-label="workoutSubmit"
          className="btn btn-outline-secondary"
          disabled={false}
          form="addWorkout"
          type="submit"
        >
          Submit Workout
        </button>
      </div>
    </div>
  </div>
</div>
`;
